### PR TITLE
Update lifecycle functions

### PR DESCRIPTION
**Pull request by dthuering**
ported changes from https://github.com/Ashoat/squadcal/blob/master/native/components/tag-input.react.js for updating React lifecycle methods.

### CODE REVIEW

**Comment by dthuering** [comment]
Hope, I ported all the changes.

File: index.js [hunk 1]
```diff
@@ -23,7 +23,7 @@ import invariant from 'invariant';
 const windowWidth = Dimensions.get('window').width;
 
 type KeyboardShouldPersistTapsProps =
-  "always" | "never" | "handled" | false | true;
+"always" | "never" | "handled" | false | true;
```
Comment (Ashoat) [inline index.js:26]: Can you undo this spacing change?

File: index.js [hunk 2]
```diff
@@ -94,13 +94,14 @@ type OptionalProps = {
   onHeightChange?: (height: number) => void,
   /**
    * Any ScrollView props (horizontal, showsHorizontalScrollIndicator, etc.)
-  */
+   */
```
Comment (Ashoat) [inline index.js:97]: Undo spacing

File: index.js [hunk 3]
```diff
@@ -120,6 +121,7 @@ class TagInput<T> extends React.PureComponent<Props<T>, State> {
     inputColor: PropTypes.string,
     inputProps: PropTypes.shape(TextInput.propTypes),
     maxHeight: PropTypes.number,
+    minHeight: PropTypes.number,
```
Comment (Ashoat) [inline index.js:124]: You're introducing a new prop here, but you haven't documented it in the `README`, and there's no corresponding Flow update for it. I think it's best to keep this new prop to a separate PR and limit this PR to just the fixes for the React lifecycle deprecation issue

File: index.js [hunk 4]
```diff
@@ -280,18 +273,18 @@ class TagInput<T> extends React.PureComponent<Props<T>, State> {
             <View style={styles.tagInputContainer}>
               {tags}
               <View style={[
-                styles.textInputContainer,
-                { width: this.state.inputWidth },
-              ]}>
+                  styles.textInputContainer,
```
Comment (Ashoat) [inline index.js:276]: Undo spacing change

**Review by Ashoat — CHANGES_REQUESTED** [review]
There's a lot of stuff here that seems unrelated to the React lifecycle method deprecation issue. The spacing changes and the new prop were ported from my fork, but aren't necessary to address the issue. I'm open to taking the new prop, but it would be best to keep it to a separate PR.

Outside of that, I looked back at the [changes](https://github.com/Ashoat/squadcal/commit/507c63dad8f30162ab1804b3a58171a1e3c09f93#diff-3ea50173114fbfdbcbeb8962820c8be5) I made in my project and it looks like I redesigned the data flow. It looks like you took a similar approach here, but I don't currently have a full grasp of the changes, so it would helpful if you could explain the new philosophy and how it maps to the old.

Sorry I'm making it so complicated! It may be easier to simplify this PR and just fix the component lifecycle deprecation issue without redesigning the data flow.

**Pull request closed without merging**